compiler/hash-intrinsics/src/intrinsics.rs

```diff
@@ -0,0 +1,123 @@
+//! Definition and lookup of intrinsics.
+use std::{fmt::Debug, process};
+
+use hash_source::identifier::Identifier;
+use hash_types::new::{
+    environment::env::AccessToEnv,
+    fns::{FnBody, FnDef, FnDefId, FnTy},
+    intrinsics::IntrinsicId,
+    mods::{ModMemberData, ModMemberValue},
+    terms::TermId,
+    utils::common::CommonUtils,
+};
+use hash_utils::store::{DefaultPartialStore, PartialCloneStore, PartialStore, Store};
+
+use crate::{primitives::AccessToPrimitives, utils::PrimitiveUtils};
+
+/// Information about an intrinsic.
+///
+/// An intrinsic contains an ID which is just a symbol internally, a
+/// corresponding function definition (in order to be able to call it from the
+/// code), and an implementation which is a Rust function that is called when
+/// the intrinsic is called.
+pub struct Intrinsic {
+    /// The ID of the intrinsic.
+    pub id: IntrinsicId,
+    /// The function definition of the intrinsic.
+    pub fn_def: FnDefId,
+    /// The implementation of the intrinsic.
+    pub implementation: fn(&(dyn AccessToPrimitives), &[TermId]) -> TermId,
```

Comment:
who is this intended to be used by, presumably the evaluation part of tc?

Reply:
yeah. will add